pyccel/codegen/printing/cucode.py

```diff
@@ -131,4 +134,44 @@
                           global_variables,
                           function_declaration,
                           "#endif // {name.upper()}_H\n"))
+    def _print_Allocate(self, expr):
+        variable = expr.variable
+        if not isinstance(variable.class_type, CudaArrayType):
+            return super()._print_Allocate(expr)
+        shape = ", ".join(self._print(i) for i in expr.shape)
+        if isinstance(variable.class_type, CudaArrayType):
+            dtype = self.find_in_ndarray_type_registry(variable.dtype)
+        else:
+            raise NotImplementedError(f"Don't know how to index {variable.class_type} type")
+        shape_Assign = f"int64_t shape_Assign_{expr.variable.name} [] = {{{shape}}};\n"
+
+        is_view = 'false' if variable.on_heap else 'true'
+        memory_location = variable.class_type.memory_location
+        if memory_location in ('device', 'host'):
+            memory_location = 'allocateMemoryOn' + str(memory_location).capitalize()
+        else:
+            memory_location = 'managedMemory'
```

Comment:
/bot accept
For the moment, it's in good condition. So if the memory location is not device and host, I set it as managed, and we will add this feature soon

Reply:
I am reassured that this code is never reached but I don't think it should be needed until we want to add managed memory. If/When we do add managed memory this may not be the strategy taken to handle. For example Pyccel could use context to determine where to place the synchronisation calls.

As you don't know what we will do in the future I would rather you did not add code related to managed memory